replica/database.cc

```diff
@@ -2848,8 +2865,8 @@ future<rwlock::holder> database::tables_metadata::hold_write_lock() {
     co_return co_await _cf_lock.hold_write_lock();
 }
 
-future<> database::tables_metadata::add_table(database& db, keyspace& ks, table& cf, schema_ptr s) {
-    auto holder = co_await _cf_lock.hold_write_lock();
+void database::tables_metadata::add_table(database& db, keyspace& ks, table& cf, schema_ptr s) {
+    SCYLLA_ASSERT(!_cf_lock.try_write_lock()); // lock should be acquired before the call
```

Comment:
Ideally an `assert` should not make any changes. why don't use `_cf_lock.locked()` here?

Reply:
hm, I think it would pass such check even if only read lock is held, no?

Reply:
Yes, it will check if any lock is held. Then it would be better to extend the `basic_semaphore` interface.